tests/unit/src/test/scala/tests/OverridesSuite.scala

```diff
@@ -0,0 +1,157 @@
+package tests
+
+import com.sourcegraph.lsif_semanticdb.Symtab
+
+import scala.meta.Input
+import munit.{FunSuite, TestOptions}
+
+class OverridesSuite extends FunSuite with TempDirectories {
+
+  val targetroot = new DirectoryFixture()
+
+  override def munitFixtures: Seq[Fixture[_]] =
+    super.munitFixtures ++ List(targetroot)
+
+  def checkOverrides(
+      options: TestOptions,
+      source: String,
+      extractSymbol: String,
+      fn: (Int, Array[String]) => Unit,
+      qualifiedClassName: String = "example.Parent"
+  ): Unit = {
+    test(options) {
+      val compiler = new TestCompiler(targetroot())
+      val relativePath = qualifiedClassName.replace('.', '/') + ".java"
+      val input = Input.VirtualFile(relativePath, source)
+      val result = compiler.compileSemanticdb(List(input))
+      val symtab = new Symtab(result.textDocument)
+
+      val count = symtab.symbols.get(extractSymbol).getOverriddenSymbolsCount
+      val syms = symtab.symbols.get(extractSymbol).getOverriddenSymbolsList.toArray(new Array[String](count))
+      fn(count, syms)
+    }
+  }
+
+  checkOverrides(
+    "same file",
+    """package example;
+      |
+      |class Parent {
+      | public void stuff() {}
+      |
+      | class Child extends Parent {
+      |   @Override
+      |   public void stuff() {}
+      | }
+      |}
+      |""".stripMargin,
+    "example/Parent#Child#stuff().",
+    { case (count, symbols) =>
+      assertEquals(count, 1)
+      assertEquals(symbols.apply(0), "example/Parent#stuff().")
+    }
+  )
+
+  checkOverrides(
+    "external override",
+    """package example;
+      |
+      |class Parent {
+      | @Override
+      | public String toString() { return ""; }
+      |}
+      |""".stripMargin,
+    "example/Parent#toString().",
+    { case (count, symbols) =>
+      assertEquals(count, 1)
+      assertEquals(symbols.apply(0), "java/lang/Object#toString().")
+    }
+  )
+
+  checkOverrides(
+    "external override",
+    """package example;
+      |
+      |class Parent {
+      | interface Test {
+      |   public void stuff();
+      | }
+      |
+      | class Child implements Test {
+      |   @Override
+      |   public void stuff() {}
+      | }
+      |}
+      |""".stripMargin,
+    "example/Parent#Child#stuff().",
+    { case (count, symbols) =>
+      assertEquals(count, 1)
+      assertEquals(symbols.apply(0), "example/Parent#Test#stuff().")
+    }
+  )
+
+  checkOverrides(
+    "type params",
+    """package example;
+      |
+      |class Parent {
+      | interface Haha<T> {
+      |   void add(T elem);
+      | }
+      | class IntHaha implements Haha<Integer> {
+      |  void add(Integer elem) {}
+      | }
+      |}
+      |""".stripMargin,
+    "example/Parent#IntHaha#add().",
+    {
+      case (count, symbols) =>
+        assertEquals(count, 1)
+        assertEquals(symbols.apply(0), "example/Parent#Haha#add().")
```

Comment:
👨‍🍳 💋

Reply:
This is surprisingly difficult to implement if you roll your own semantic analysis